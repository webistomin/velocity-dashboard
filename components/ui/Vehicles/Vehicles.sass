+b('vehicles')
  +e('grid')
    display: grid
    justify-content: start
    grid-template-columns: repeat(12, 1fr)
    grid-gap: rem(14px)

    +media($screen-sm)
      grid-gap: rem(24px)

    +media($screen-lg)
      grid-gap: rem(34px)

  +e('block')
    &:first-of-type
      grid-column: 12 span
      grid-row: 1 span

    &:nth-of-type(2)
      grid-column: 12 span
      grid-row: 2 span

    &:nth-of-type(3)
      grid-column: 12 span
      grid-row: 4 span

    +media($screen-lg)
      &:first-of-type
        grid-column: 8 span
        grid-row: 1 span

      &:nth-of-type(2)
        grid-column: 4 span
        grid-row: 6 span

      &:nth-of-type(3)
        grid-column: 8 span
        grid-row: 8 span

  +e('counter')
    color: var(--color-gray-3)
    font: var(--text-paragraph)
    padding-left: rem(12px)

  +e('filter-group')
    &:not(:last-of-type)
      margin-bottom: rem(30px)

  +e('filters')
    margin-top: rem(24px)

  +e('filter-block')
    &.base-block__content
      justify-content: flex-start


