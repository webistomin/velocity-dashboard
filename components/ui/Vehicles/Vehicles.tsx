import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import BaseBlock from 'components/base/BaseBlock';
import BaseTitle from 'components/base/BaseTitle';
import BaseTable from 'components/base/BaseTable';
import BaseButton from 'components/base/BaseButton';
import BaseSlider from 'components/base/BaseSlider';
import BaseSelect from 'components/base/BaseSelect';
// @ts-ignore
import { CollapseTransition } from 'vue2-transitions';

import './Vehicles.sass';

@Component({
  name: 'Vehicles',
})
export default class Vehicles extends VueComponent {
  public isFiltersVisible: boolean = false;

  public tableData = [
    {
      id: '#2178',
      subject: 'Refund request',
      date: '05/04/2018',
      status: 'Active',
    },
    {
      id: '#2179',
      subject: 'Active request',
      date: '05/04/2018',
      status: 'Active',
    },
    {
      id: '#2180',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2181',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2182',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2183',
      subject: 'Refund request',
      date: '05/11/2018',
      status: 'Active',
    },
    {
      id: '#2184',
      subject: 'Active request',
      date: '05/03/2018',
      status: 'Active',
    },
    {
      id: '#2185',
      subject: 'Refund request',
      date: '02/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2186',
      subject: 'Refund request',
      date: '05/04/2019',
      status: 'Resolved',
    },
    {
      id: '#2187',
      subject: 'Refund request',
      date: '01/04/2019',
      status: 'Unresolved',
    },
  ];

  public tableConfig = [
    {
      key: 'id',
      title: 'Id',
    },
    {
      key: 'subject',
      title: 'Subject',
    },
    {
      key: 'date',
      title: 'Latest update',
    },
    {
      key: 'status',
      title: 'Status',
    },
  ];

  public slider1: number = 0;
  public slider2: number = 0;

  public select1: string = '';
  public select2: string = '';
  public select3: string = '';

  public toggleFiltersVisibility(): void {
    this.isFiltersVisible = !this.isFiltersVisible;
  }

  public mounted(): void {
    if (window.matchMedia('(min-width: 1024px)').matches) {
      this.isFiltersVisible = true;
    }
  }

  public render(): VNode {
    return (
      <section class='vehicles'>
        <div class='vehicles__container container'>
          <div class='vehicles__grid'>
            <BaseBlock class='vehicles__block' simple={true}>
              <div class='vehicles__heading'>
                <BaseTitle level={3}>
                  Vehicles Dashboard
                  <sup class='vehicles__counter'>1192 Total</sup>
                </BaseTitle>
              </div>
            </BaseBlock>
            <BaseBlock class='vehicles__block' title='filter' contentMix='vehicles__filter-block'>
              <BaseButton class='vehicles__filters-toggle' theme='light' onClick={() => this.toggleFiltersVisibility()}>
                {this.isFiltersVisible ? 'Hide' : 'Show'} filter
              </BaseButton>
              <CollapseTransition>
                <div class='vehicles__filters' vShow={this.isFiltersVisible}>
                  <div class='vehicles__filter-group'>
                    <BaseSlider
                      value={this.slider1}
                      onChange={(value: number) => (this.slider1 = value)}
                      min={0}
                      max={753}
                      labelStart='Trips taken'
                      labelEnd='753'
                    />
                  </div>
                  <div class='vehicles__filter-group'>
                    <BaseSlider
                      value={this.slider2}
                      onChange={(value: number) => (this.slider2 = value)}
                      min={0}
                      max={14}
                      labelStart='Trips taken'
                      labelEnd='14 days'
                    />
                  </div>
                  <div class='vehicles__filter-group'>
                    <BaseSelect
                      label='Vehicle model'
                      id='vehicles-model-select'
                      value={this.select1}
                      options={['1', '2', '3']}
                      onInput={(value: string) => (this.select1 = value)}
                      placeholder='Vehicle model'
                    />
                  </div>
                  <div class='vehicles__filter-group'>
                    <BaseSelect
                      label='Status'
                      id='vehicles-status-select'
                      value={this.select2}
                      options={['1', '2', '3']}
                      onInput={(value: string) => (this.select2 = value)}
                      placeholder='Status'
                    />
                  </div>
                  <div class='vehicles__filter-group'>
                    <BaseSelect
                      label='Location'
                      id='vehicles-location-select'
                      value={this.select3}
                      options={['1', '2', '3']}
                      onInput={(value: string) => (this.select3 = value)}
                      placeholder='Location'
                    />
                  </div>
                </div>
              </CollapseTransition>
            </BaseBlock>
            <BaseBlock simple={true} class='vehicles__block' contentMix='vehicles__filter-block'>
              <BaseTable tableConfig={this.tableConfig} tableData={this.tableData} />
            </BaseBlock>
          </div>
        </div>
      </section>
    );
  }
}
