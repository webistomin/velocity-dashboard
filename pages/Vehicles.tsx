import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import Vehicles from 'components/ui/Vehicles';

@Component({
  name: 'VehiclesPage',
  middleware: ['auth-guard', 'auth-admin'],
})
export default class VehiclesPage extends VueComponent {
  render(): VNode {
    return (
      <main class='page-content page-content_offset'>
        <Vehicles />
      </main>
    );
  }
}
