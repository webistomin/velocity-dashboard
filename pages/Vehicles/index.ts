import Vehicles from './Vehicles';

export default Vehicles;
