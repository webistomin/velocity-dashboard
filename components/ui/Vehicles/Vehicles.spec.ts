import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vehicles from './Vehicles';

describe('Vehicles UI', () => {
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();

  const factory = (propsData = {}, options = {}) => {
    return shallowMount(Vehicles, {
      localVue,
      router,
      propsData: {
        isNavOpened: false,
        ...propsData,
      },
      ...options,
    });
  };

  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => {
        return {
          matches: true,
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
        };
      }),
    });
  });

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
