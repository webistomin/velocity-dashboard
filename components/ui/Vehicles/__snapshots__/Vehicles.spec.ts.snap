// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles UI Matches snapshot 1`] = `
"<section class=\\"vehicles\\">
  <div class=\\"vehicles__container container\\">
    <div class=\\"vehicles__grid\\">
      <baseblock-stub issimple=\\"true\\" class=\\"vehicles__block\\">
        <div class=\\"vehicles__heading\\">
          <vuecomponent-stub level=\\"3\\">Vehicles Dashboard</vuecomponent-stub>
        </div>
      </baseblock-stub>
      <baseblock-stub title=\\"filter\\" contentmix=\\"vehicles__filter-block\\" class=\\"vehicles__block\\">
        <basebutton-stub type=\\"button\\" icon=\\"\\" theme=\\"light\\" class=\\"vehicles__filters-toggle\\">Show filter</basebutton-stub>
        <collapse-transition-stub duration=\\"300\\" delay=\\"0\\" tag=\\"span\\" origin=\\"\\" styles=\\"[object Object]\\">
          <div class=\\"vehicles__filters\\" style=\\"display: none;\\"></div>
        </collapse-transition-stub>
      </baseblock-stub>
      <baseblock-stub contentmix=\\"vehicles__filter-block\\" issimple=\\"true\\" class=\\"vehicles__block\\">
        <vuecomponent-stub text=\\"No data was provided\\"></vuecomponent-stub>
      </baseblock-stub>
    </div>
  </div>
</section>"
`;
